jest.config.mjs

```diff
@@ -0,0 +1,46 @@
+// eslint-disable-next-line import/extensions
+import jestPuppeteerConfig from './jest-puppeteer.config.js';
+
+const { headless } = jestPuppeteerConfig.launch;
+
+export default {
+  maxWorkers: headless
+    ? '50%'
+    : 1,
+
+  projects: [
+    {
+      displayName: 'unit tests',
+      setupFilesAfterEnv: ['<rootDir>/lib/jest-helpers.js'],
+      clearMocks: true,
+      testMatch: [
+        '**/*test.js',
+        '!**/*jsdom.test.js',
+        '!**/*browser.test.js',
+      ],
+    },
+    {
+      displayName: 'jsdom tests',
+      testEnvironment: 'jsdom',
+      snapshotSerializers: ['jest-serializer-html'],
+      setupFilesAfterEnv: ['<rootDir>/lib/jest-helpers.js'],
+      clearMocks: true,
+      testMatch: [
+        '**/*jsdom.test.js',
+      ],
+    },
+    {
+      displayName: 'browser tests',
+      preset: 'jest-puppeteer',
+      clearMocks: true,
+      setupFilesAfterEnv: ['expect-puppeteer', '<rootDir>/lib/browser-tests/jest-setup.js'],
+      testMatch: [
+        '**/*browser.test.js',
+      ],
+    },
+  ],
```

Comment:
previously the environment of a test was configured by comments at the top of test files, there were some limits of what custom options you could specify that way so instead I've opted to rename tests with different suffixes. This matches the approach that govuk-frontend uses too https://github.com/alphagov/govuk-frontend/blob/main/jest.config.mjs